src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -284,6 +298,32 @@ public void Dispose()
             Dispose(true);
         }
 
+        internal bool IsEnoughMemoryToScale(long hostProcessSize, IEnumerable<long> workerChannelSizes, long memoryLimit)
+        {
+            if (memoryLimit <= 0)
+            {
+                return true;
+            }
+
+            // Checking memory before adding a new worker
+            // By adding `maxWorkerSize` to current memeory consumption we are predicting what will be overall memory consumption after adding a new worker.
+            // We do not want this value to be more then 80%.
+            long maxWorkerSize = workerChannelSizes.Max();
+            long currentMemoryConsumption = workerChannelSizes.Sum() + hostProcessSize;
+            if (currentMemoryConsumption + maxWorkerSize > memoryLimit * 0.8)
+            {
+                var message = new
+                {
+                    TotalMemory = memoryLimit,
+                    MaxWorkerSize = maxWorkerSize,
+                    CurrentMemoryConsumption = currentMemoryConsumption
+                };
+                _logger.LogDebug("Starting new language worke canceled: {@message}", message);
```

Comment:
What logger providers do we use? The default `Microsoft.Extensions.Logging` logger does not support object deconstruction in logs. So you can't use `@` there. If we use like Serilog or something else, that does support it.

Reply:
Somehow `TestLogger `works but production logger does not work - moving back to to proviso approach